src/org/zaproxy/zap/extension/exportreport/Export/ReportExport.java

```diff
@@ -0,0 +1,315 @@
+		String otherinfo = Constant.messages.getString("exportreport.details.otherinfo.label");
+		String reference = Constant.messages.getString("exportreport.details.reference.label");
+		String cweid = Constant.messages.getString("exportreport.details.cweid.label");
+		String wascid = Constant.messages.getString("exportreport.details.wascid.label");
+		String requestheader = Constant.messages.getString("exportreport.details.requestheader.label");
+		String responseheader = Constant.messages.getString("exportreport.details.responseheader.label");
+		String requestbody = Constant.messages.getString("exportreport.details.requestbody.label");
+		String responsebody = Constant.messages.getString("exportreport.details.responsebody.label");
+
+		try {
+			for (int i = 0; i < siteNumber; i++) {
+				SiteNode site = (SiteNode) root.getChildAt(i);
+				String siteName = ScanPanel.cleanSiteName(site, true);
+				String[] hostAndPort = siteName.split(":");
+				boolean isSSL = (site.getNodeName().startsWith("https"));
+
+				Sites s = new Sites();
+				s.setHost(entityEncode(hostAndPort[0]));
+				s.setName(entityEncode(site.getNodeName()));
+				s.setPort(entityEncode(hostAndPort[1]));
+				s.setSSL(String.valueOf(isSSL));
+
+				List<Alert> alerts = site.getAlerts();
+				Alerts a = new Alerts();
+				String temp = "";
+				for (Alert alert : alerts) {
+					
+					if(!alertSeverity.contains(Alert.MSG_RISK[alert.getRisk()])) {
+						continue;
+					}
+
+					AlertItem item = new AlertItem();
+					item.setPluginID(entityEncode(Integer.toString(alert.getPluginId())));
+					item.setAlert(entityEncode(alert.getAlert()));
+					item.setRiskCode(entityEncode(Integer.toString(alert.getRisk())));
+					item.setRiskDesc(entityEncode(Alert.MSG_RISK[alert.getRisk()]));
+					item.setConfidence(entityEncode(Alert.MSG_CONFIDENCE[alert.getConfidence()]));
+
+					for (int j = 0; j < alertDetails.size(); j++) {
+						if (alertDetails.get(j).equalsIgnoreCase(description))
+							item.setDesc(entityEncode(alert.getDescription()));
+						if (alertDetails.get(j).equalsIgnoreCase(solution))
+							item.setSolution(entityEncode(alert.getSolution()));
+						if (alertDetails.get(j).equalsIgnoreCase(otherinfo) && alert.getOtherInfo() != null
+								&& alert.getOtherInfo().length() > 0) {
+							item.setOtherInfo(entityEncode(alert.getOtherInfo()));
+						}
+						if (alertDetails.get(j).equalsIgnoreCase(reference))
+							item.setReference(entityEncode(alert.getReference()));
+						if (alertDetails.get(j).equalsIgnoreCase(cweid) && alert.getCweId() > 0)
+							item.setCWEID(entityEncode(Integer.toString(alert.getCweId())));
+						if (alertDetails.get(j).equalsIgnoreCase(wascid))
+							item.setWASCID(entityEncode(Integer.toString(alert.getWascId())));
+
+						temp = alert.getMessage().getRequestHeader().toString();
+						if (alertDetails.get(j).equalsIgnoreCase(requestheader) && temp != null	&& temp.length() > 0) {
```

Comment:
@thc202 It was in some cases when i was testing, that's why i added it because it was throwing me errors when the response was `null`, after that i added it to the requests as well just to be safe.